src/jobs/models/taskManager.ts

```diff
@@ -37,42 +47,57 @@ export class TaskManager {
   }
 
   public async createTask(req: CreateTasksRequest): Promise<CreateTasksResponse> {
-    const repo = await this.getRepository();
     const jobId = Array.isArray(req) ? req[0].jobId : req.jobId;
     this.logger.log('info', `creating task(s) for job ${jobId}`);
-    const res = await repo.createTask(req);
+    const res = await this.repository.createTask(req);
     return res;
   }
 
   public async getTask(req: ISpecificTaskParams): Promise<IGetTaskResponse> {
-    const repo = await this.getRepository();
-    const res = await repo.getTask(req);
+    const res = await this.repository.getTask(req);
     if (res === undefined) {
       throw new EntityNotFound('Task not found');
     }
     return res;
   }
 
   public async updateTask(req: IUpdateTaskRequest): Promise<void> {
-    const repo = await this.getRepository();
     this.logger.log('info', `updating task ${req.taskId} from job ${req.jobId}`);
-    await repo.updateTask(req);
+    await this.repository.updateTask(req);
   }
 
   public async deleteTask(req: ISpecificTaskParams): Promise<void> {
-    const repo = await this.getRepository();
     this.logger.log('info', `deleting task ${req.taskId} from job ${req.jobId}`);
-    const res = await repo.deleteTask(req);
+    const res = await this.repository.deleteTask(req);
     return res;
   }
 
+  public async getTaskStatus(req: IAllTasksParams): Promise<IGetTasksStatus> {
+    const jobManager = container.resolve(JobManager);
+    const { version: resourceVersion, resourceId } = await jobManager.getJob(req);
+
+    const tasks = await this.repository.getTasks(req);
+    this.logger.log('info', `Getting tasks statuses for jobId ${req.jobId}`);
+    const completedTasksCount = tasks.filter((task) => task.status === OperationStatus.COMPLETED).length;
```

Comment:
all the the data calculations and stats should be done in db to improve efficiency and prevent out of memory errors